File: accsaber-database/src/main/java/de/ixsen/accsaber/database/views/extenders/WithRank.java
package de.ixsen.accsaber.database.views.extenders;

import org.hibernate.annotations.Immutable;

import javax.persistence.MappedSuperclass;

@MappedSuperclass
@Immutable
public abstract class WithRank {
    private Long ranking;
    private Double ap;

    public Long getRanking() {
        return this.ranking;
    }

    public Double getAp() {
        return this.ap;
    }
}

File: accsaber-database/src/main/kotlin/de/ixsen/accsaber/database/views/extenders/WithRank.kt
package de.ixsen.accsaber.database.views.extenders

import org.hibernate.annotations.Immutable
import javax.persistence.MappedSuperclass

@MappedSuperclass
@Immutable
abstract class WithRank {
    val ranking: Long? = null
    val ap: Double? = null
}